src/lsli/main.py

```diff
@@ -0,0 +1,449 @@
+        loggers = [logging.getLogger(config.SKID_NAME), logging.getLogger("palletjack")]
+        _remove_log_file_handlers(log_name, loggers)
+
+
+def _load_records_from_graphql(url: str, query: str, limit: int) -> pd.DataFrame:
+    """Load records from a GraphQL endpoint in chunks
+
+    Args:
+        url (str): GraphQL endpoint URL
+        query (str): GraphQL query string
+        limit (int): The max number of records to return per chunk
+
+    Returns:
+        pd.DataFrame: GraphQL records as a DataFrame
+    """
+
+    transport = RequestsHTTPTransport(
+        url=url,
+        verify=True,
+        retries=3,
+    )
+    client = Client(transport=transport, fetch_schema_from_transport=True)
+    query = gql(query)
+
+    result_length = limit
+    offset = 0
+    records = []
+
+    while result_length == limit:
+        result = client.execute(query, variable_values={"offset": offset, "limit": limit})
+        result_length = len(result["getLccrMapUGRC"])
+        module_logger.debug("Offset: %s, Length: %s", format(offset, ","), format(result_length, ","))
+        offset += limit
+        records.extend(result["getLccrMapUGRC"])
+
+    return pd.DataFrame(records)
+
+
+def _spatialize_data(df: pd.DataFrame) -> pd.DataFrame:
+    """Convert a dataframe to a spatially-enabled dataframe accounting for both WGS84 and UTM NAD83 coordinates
+
+    Any rows with latitude < 100 are assumed to be WGS84, while rows with latitude > 100 are assumed to be UTM NAD83.
+
+    Args:
+        df (pd.DataFrame): Input Dataframe with "latitude" and "longitude" columns
+
+    Returns:
+        pd.DataFrame: Spatially-enabled DataFrame in Web Mercator (EPSG:3857)
+    """
+
+    web_mercator_dfs = []
+
+    wgs_data = df[df["latitude"] < 100]
+    if not wgs_data.empty:
+        module_logger.debug("Loading %s rows with WGS84 coordinates", format(len(wgs_data), ","))
+        wgs_spatial = pd.DataFrame.spatial.from_xy(wgs_data, "longitude", "latitude", sr=4326)
+        module_logger.debug("Projecting WGS84 data to Web Mercator")
+        wgs_spatial.spatial.project(3857)
+        web_mercator_dfs.append(wgs_spatial)
+
+    utm_data = df[df["latitude"] > 100]
+    if not utm_data.empty:
+        module_logger.debug("Loading %s rows with UTM coordinates", format(len(utm_data), ","))
+        utm_spatial = pd.DataFrame.spatial.from_xy(utm_data, "longitude", "latitude", sr=26912)
+        module_logger.debug("Projecting UTM data to Web Mercator")
+        utm_spatial.spatial.project(3857)
+        web_mercator_dfs.append(utm_spatial)
+
+    return pd.concat(web_mercator_dfs)
+
+
+class GoogleSheetData:
+    """Represents data about whole systems loaded from a Google Sheet"""
+
+    systems = pd.DataFrame()
+    links = pd.DataFrame()
+    cleaned_systems_dataframe = pd.DataFrame()
+    cleaned_water_service_areas = pd.DataFrame()
+    final_systems = pd.DataFrame()
+
+    missing_geometries = {}
+    invalid_pwsids = []
```

Comment:
python is weird about initializing default values to mutable data outside of a constructor. This can lead to unexpected shared memory and bugs between instances of this class. It's generally speaking safer to initialize this array in the init if it's not intended to be shared.

Maybe this example will describe the issue a bit better

```py
class MyClass:
    my_list = []  # Initialized outside the constructor

    def __init__(self):
        pass

obj1 = MyClass()
obj2 = MyClass()

obj1.my_list.append(1)

print(obj1.my_list)  # Output: [1]
print(obj2.my_list)  # Output: [1] - Unexpected!
```